analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -308,3 +340,25 @@ internal fun Identifier.toVcsInfo(): VcsInfo {
     val vcsInfo = VcsHost.toVcsInfo("https://$name")
     return vcsInfo.copy(revision = getRevision(version))
 }
+
+/**
+ * Parse the given [output] generated by the _go mod why_ command and the names of packages that are reported to be
+ * used by the main module.
+ * See https://golang.org/ref/mod#go-mod-why.
+ */
+internal fun parseWhyOutput(output: String): Set<String> {
+    val usedPackages = mutableSetOf<String>()
+    var currentPackage: String? = null
+
+    output.lineSequence().forEach { line ->
+        if (line.startsWith(PACKAGE_SEPARATOR)) {
+            currentPackage = line.removePrefix(PACKAGE_SEPARATOR)
```

Comment:
I have to admit, personally I liked the original code better then, as it was not less readable IMO, but avoided the duplicate `startsWith()`. But I'll leave it to you guys, I'm just throwing in my comments 😉

Reply:
Switched back to the implementation using `substring()`; I liked it better, too.